lib/ohai/plugins/shard.rb

```diff
@@ -34,6 +33,27 @@ def default_sources
     [:machinename, :serial, :uuid]
   end
 
+  def default_digest_algorithm
+    if defined?(OpenSSL.fips_mode) && OpenSSL.fips_mode
```

Comment:
Is there a reason you're not just using the data we already gather from the FIPS plugin? We gather data in very specific ways within that plugin per platform.

Reply:
No specific reason, this seemed more accurate. It was only added in Ruby 2.5 which is why I'm checking `defined?` and probably why we didn't use it before :)